src/scripts/hubot-jenkins-notifier.coffee

```diff
@@ -115,24 +144,51 @@ class JenkinsNotifier
     [@dataMethodJSONParse, @dataMethodRaw].forEach(filterChecker)
 
     if !data || typeof data.build != 'object'
-      @error new Error("Unable to process data"), req.body
+      @error new Error("Unable to process data - data empty or not an object"), req.body
       return
+
+    @logMessage(query.trace, data.build)
+
     fullurl = data.build.full_url || data.build.url
 
+    if data.build.phase in ['COMPLETED']
+      @logMessage(query.trace, "Ignoring phase COMPLETED")
+      return
+
+    @logMessage(query.trace, "#{data.name} #{data.build.phase} #{data.build.status}")
+
+    if data.build.phase in ['STARTED']
+      @robot.send envelope, "#{data.name} build ##{data.build.number} started: #{fullurl}"
```

Comment:
Some people only want to know whats happening with a build when something
is broken.

Started will be based on the last build's outcome.

notStrat=
s = only notify on start if last job was a failure
S = always notify on start

On Mon, Aug 22, 2016 at 7:10 AM, Jens Doose <notifications@github.com>
wrote:

> In src/scripts/hubot-jenkins-notifier.coffee
> <https://github.com/halkeye/hubot-jenkins-notifier/pull/7#discussion_r75684234>
> :
>
> >      fullurl = data.build.full_url || data.build.url
> >
> > +    if data.build.phase in ['COMPLETED']
> > +      @logMessage(query.trace, "Ignoring phase COMPLETED")
> > +      return
> > +
> > +    @logMessage(query.trace, "#{data.name} #{data.build.phase} #{data.build.status}")
> > +
> > +    if data.build.phase in ['STARTED']
> > +      @robot.send envelope, "#{data.name} build ##{data.build.number} started: #{fullurl}"
>
> ;) Yes you did.
> I wasn't so sure about this, since shouldNotify respects the current and
> the previous build outcome. But STARTED does not have an outcome, right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/halkeye/hubot-jenkins-notifier/pull/7/files/ebbd9282c7440d21e2e2fdb107daf33d10d2e09a#r75684234>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGuB4pasc0kjE4wjP2FsWNFEp1gOFomks5qia3ygaJpZM4JY0mC>
> .
>